### PR TITLE
annotate MissingFile errors

### PR DESCRIPTION
**Pull request by Profpatsch**
Should make it easier to debug complex dhall configurations.

Implements https://github.com/dhall-lang/dhall-haskell/issues/189#issuecomment-350789514

### CODE REVIEW

File: src/Dhall/Import.hs [hunk 1]
```diff
@@ -589,7 +598,7 @@ exprFromPath m (Path {..}) = case pathType of
                 exists <- Filesystem.isFile path
                 if exists
                     then return ()
-                    else Control.Exception.throwIO MissingFile
+                    else Control.Exception.throwIO (MissingFile path)
```
Comment (Profpatsch) [inline src/Dhall/Import.hs]: Is `path` already canonicalized here?

File: src/Dhall/Import.hs [hunk 2]
```diff
@@ -134,6 +134,7 @@ import Data.Text.Lazy.Builder (Builder)
 import Data.Traversable (traverse)
 #endif
 import Data.Typeable (Typeable)
+import qualified Filesystem.Path.CurrentOS
```
Comment (Gabriel439) [inline src/Dhall/Import.hs]: For alignment reasons I try to keep all the qualified imports together at the bottom

File: src/Dhall/Import.hs [hunk 3]
```diff
@@ -176,11 +178,11 @@ import qualified Dhall.Parser
 import qualified Dhall.Context
 import qualified Dhall.TypeCheck
 import qualified Filesystem
-import qualified Filesystem.Path.CurrentOS
 import qualified Network.HTTP.Client              as HTTP
 import qualified Network.HTTP.Client.TLS          as HTTP
 import qualified Filesystem.Path.CurrentOS        as Filesystem
 import qualified System.Environment
+import qualified System.Directory
```
Comment (Gabriel439) [inline src/Dhall/Import.hs]: What's the reason for the `System.Directory` import?  It seems like it's not used

File: src/Dhall/Import.hs [hunk 4]
```diff
@@ -301,15 +303,22 @@ instance Show PrettyHttpException where
...
             "\n"
-        <>  "\ESC[1;31mError\ESC[0m: Missing file\n"
+        <>  "\ESC[1;31mError\ESC[0m: Missing file "
+        <>  Data.Text.unpack formattedPath
+        <>  "\n"
+      where
+        formattedPath = case Filesystem.Path.CurrentOS.toText path of
+            (Right t) -> t
+            (Left  t) -> t
+                <> "\n(filename contains non-displayable characters)"
```
Comment (Gabriel439) [inline src/Dhall/Import.hs]: Minor nitpick: Perhaps use: `\ESC[1;31mWarning\ESC[0m: Filename contains ...`

File: dhall.cabal [hunk 5]
```diff
@@ -100,6 +100,7 @@ Library
         containers           >= 0.5.0.0  && < 0.6 ,
         contravariant                       < 1.5 ,
         cryptohash                          < 0.12,
+        directory            >= 1.2.2.0  && < 1.4 ,
```
Comment (Gabriel439) [inline dhall.cabal]: This import looks like it is unused

File: src/Dhall/Import.hs (hunk 3 above)
Comment (Profpatsch) [reply src/Dhall/Import.hs]: Oh right, I played around with `makeAbsolute` and forgot to remove it once I saw that it leads nowhere.

File: src/Dhall/Import.hs (hunk 1 above)
Comment (Gabriel439) [reply src/Dhall/Import.hs]: Yes.  The part where `exprFromPath` gets called (very indirectly) is here: https://github.com/Profpatsch/dhall-haskell/blob/8b741bae1fba6c7a007949db7f5e8058b5adabc7/src/Dhall/Import.hs#L752

File: src/Dhall/Import.hs (hunk 2 above)
Comment (Profpatsch) [reply src/Dhall/Import.hs]: Fixed in patch.

**Comment by Profpatsch** [comment]
Should have addressed your comments.

**Comment by Gabriel439** [comment]
Awesome!  I'll merge once CI is green.  Thanks for doing this :)

**Pull request merged**